### PR TITLE
TestSchedulingQueue: Remove the unnecessary slice and for loop

### PR DESCRIPTION
**Pull request by gaurav1086**
Signed-off-by: Gaurav Singh <gaurav1086@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
 /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Cleanup: Remove the unnecessary slice and for loop.

**Which issue(s) this PR fixes**:

Fixes #
N/A
**Special notes for your reviewer**:
A simulated example to expose the bug: https://play.golang.com/p/vuo6rfy68sK 
Uncommenting line 24 in this example fixes the issue.
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
could you remove the loop instead?

**Comment by gaurav1086** [comment]
> could you remove the loop instead?

@alculquicondor , Yeah, currently we only have one element in the array: 

```
{
		{
			name:        "PriorityQueue close",
			q:           NewPriorityQueue(newDefaultQueueSort()),
			expectedErr: fmt.Errorf(queueClosed),
		},
	}
```
But we can leave it for more tests to be added to this queue in future. Thoughts ?

**Comment by alculquicondor** [comment]
We don't need more tests for closing the queue.

**Comment by gaurav1086** [comment]
> We don't need more tests for closing the queue.

Ah yes, didn't realize that. Remove the for loop and reverted my previous change.

**Comment by Huang-Wei** [comment]
@gaurav1086 I don't think current test would have racing conditions. It's because in each subtest, `wg.Wait()` would wait until that subtest to finish. (actually even without wg.Wait(), Golang by default run subtest one after another).

This is different with the sample code (https://play.golang.com/p/vuo6rfy68sK) you wrote - in that case, all goroutines in the `for` looping are actually running simultaneously, holding the same `test` pointer.

**Comment by alculquicondor** [comment]
> gaurav I don't think current test would have racing conditions. It's because in each subtest, `wg.Wait()` would wait until that subtest to finish. (actually even without wg.Wait(), Golang by default run subtest one after another).

That is true, but I'm welcoming the PR to remove the loop altogether.
PR description needs to be updated, though.

**Comment by gaurav1086** [comment]
> @gaurav1086 I don't think current test would have racing conditions. It's because in each subtest, `wg.Wait()` would wait until that subtest to finish. (actually even without wg.Wait(), Golang by default run subtest one after another).
> 
> This is different with the sample code (https://play.golang.com/p/vuo6rfy68sK) you wrote - in that case, all goroutines in the `for` looping are actually running simultaneously, holding the same `test` pointer.

@Huang-Wei ah, yes. Thanks for pointing out. t.Run() is not the same as a generic lambda function. So its not a race condition.

File: pkg/scheduler/internal/queue/scheduling_queue_test.go [hunk 1]
```diff
@@ -721,24 +721,22 @@ func TestSchedulingQueue_Close(t *testing.T) {
...
-				}
-				if pod != nil {
-					t.Errorf("Expected pod nil from Pop() if queue is closed, but got: %v", pod)
-				}
-			}()
-			test.q.Close()
-			wg.Wait()
-		})
-	}
+	t.Run(test.name, func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (gaurav1086) [reply pkg/scheduler/internal/queue/scheduling_queue_test.go:724]: Changes done. Thank you.

[the remark this answers is not in the document]
Comment (alculquicondor) [reply pkg/scheduler/internal/queue/scheduling_queue_test.go:722]: just use variables instead of the struct.

**Comment by Huang-Wei** [comment]
/ok-to-test

@gaurav1086 please squash the commits.

/approve
will leave /lgtm for @alculquicondor .

File: pkg/scheduler/internal/queue/scheduling_queue_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gaurav1086) [reply pkg/scheduler/internal/queue/scheduling_queue_test.go]: @alculquicondor changes done and commit messages updated. Please check. Thanks.

File: pkg/scheduler/internal/queue/scheduling_queue_test.go [hunk 2]
```diff
@@ -710,35 +710,26 @@ func TestUnschedulablePodsMap(t *testing.T) {
...
-			}()
-			test.q.Close()
-			wg.Wait()
-		})
-	}
+	name := "PriorityQueue close"
+	q := NewPriorityQueue(newDefaultQueueSort())
+	expectedErr := fmt.Errorf(queueClosed)
+
+	t.Run(name, func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (gaurav1086) [reply pkg/scheduler/internal/queue/scheduling_queue_test.go]: Yeah, correct. Removed. Please check.

File: pkg/scheduler/internal/queue/scheduling_queue_test.go [hunk 3]
```diff
@@ -710,35 +710,22 @@ func TestUnschedulablePodsMap(t *testing.T) {
...
-				if pod != nil {
-					t.Errorf("Expected pod nil from Pop() if queue is closed, but got: %v", pod)
-				}
-			}()
-			test.q.Close()
-			wg.Wait()
-		})
-	}
+	q := NewPriorityQueue(newDefaultQueueSort())
+	expectedErr := fmt.Errorf(queueClosed)
```
Comment (alculquicondor) [inline pkg/scheduler/internal/queue/scheduling_queue_test.go:714]: nit: `wantErr`

**Review by alculquicondor — COMMENTED** [review]
/approve

please squash

File: pkg/scheduler/internal/queue/scheduling_queue_test.go (hunk 3 above)
Comment (gaurav1086) [reply pkg/scheduler/internal/queue/scheduling_queue_test.go:714]: s/expectedErr/wantErr ?

Comment (alculquicondor) [reply pkg/scheduler/internal/queue/scheduling_queue_test.go:714]: yes

**Pull request merged**